### PR TITLE
docker service override.

### PR DESCRIPTION
**Pull request by enxebre**

### CODE REVIEW

File: roles/docker/templates/docker.service.j2 [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Docker Application Container Engine
+Documentation=http://docs.docker.com
+After=docker.socket early-docker.target network.target
+Requires=docker.socket early-docker.target
+
+[Service]
+EnvironmentFile=-/run/flannel_docker_opts.env
```
Comment (tayzlor) [inline roles/docker/templates/docker.service.j2]: flannel?

File: roles/docker/tasks/main.yml [hunk 2]
```diff
@@ -1,35 +1,17 @@
...
-#- name: configure docker temporary directory
-#  sudo: yes
-#  lineinfile:
-#    dest: /etc/sysconfig/docker
-#    state: present
-#    line: 'DOCKER_TMPDIR=\"{{ docker_tmp_dir }}\"'
-#  notify:
-#    - restart docker
-
-#- name: configure docker proxy
```
Comment (tayzlor) [inline roles/docker/tasks/main.yml]: where do we handle these options now ?

File: roles/docker/tasks/main.yml [hunk 3]
```diff
@@ -1,35 +1,17 @@
 ---
 # tasks file for docker
-- name: configure docker consul dns
+- name: deploy docker service
   sudo: yes
-  lineinfile:
-    dest: /etc/sysconfig/docker
-    regexp: ^OPTIONS=
-    line: 'OPTIONS=\"--storage-driver=overlay --graph={{ docker_graph_dir }} --dns 172.17.0.1 --dns 8.8.8.8 --dns-search service.{{ consul_domain }} \"'
```
Comment (tayzlor) [inline roles/docker/tasks/main.yml]: where are these options getting handled now?

File: roles/docker/templates/docker.service.j2 (hunk 1 above)
Comment (enxebre) [inline roles/docker/templates/docker.service.j2]: This is just a copy of /usr/lib/systemd/system/docker.service as suggested in here https://coreos.com/os/docs/877.1.0/customizing-docker.html we can inject drop it settings just by adding files in ``` /etc/systemd/system/docker.service.d/ ```

File: roles/docker/tasks/main.yml (hunk 2 above)
Comment (enxebre) [inline roles/docker/tasks/main.yml]: this file is ignored by coreOS. This is handle now in  roles/docker/templates/docker.service.j2
We could override DOCKER_OPTS environment variable here /etc/systemd/system/docker.service.d/
but coreOS is giving a memory error message when using more than one flag there
https://coreos.com/os/docs/877.1.0/customizing-docker.html

**Pull request merged**